Autotest API

miksenya opened the pull request.

miksenya committed:
Создал новый класс UserBugApi для автотеста api

miksenya committed:
Создал два класса: CreateUserAPi и CreateCompanyApi для автотестов.

mober2000 commented on src/test/java/com/userbug/CreateCompanyApi.java:
```diff
@@ -0,0 +1,62 @@
+        assertNotNull(companyId, "id_company не вернулся");
+        assertTrue(companyId > 0, "id_company должен быть > 0");
+    }
+}
```
Выдели запросы в отдельные классы по аналогии с Пейдж объектом, в тестах должно быть только максимум не сколько строчек а логика в отдельных классах

mober2000 commented on src/test/java/com/userbug/CreateCompanyApi.java:
```diff
@@ -0,0 +1,62 @@
+        String owner = "owner_qa@example.com";
+        String u1 = "user1_qa@example.com";
+        String u2 = "user2_qa@example.com";
+        ensureUser(owner, "Owner QA");
```
Переменные так же должны быть в отдельных классах

mober2000 commented on src/test/java/com/userbug/UserBugApi.java:
```diff
@@ -0,0 +1,39 @@
+                .header("Content-Type", "application/json")
+                .body("""
+                        {
+                            "email": "asdasda@bk.ru",
```
В этом классе все тоже самое с переменными и методами в отдельных классах

miksenya committed:
Распределил всю логику в отдельные классы.

BaseApiTest - Общие настройки RestAssured для всех
тестов.

client - Методы на HTTP вызовы:
UserClient — doRegister(...), ensureUser(...), createUserWithTasks(...). CompanyClient — createCompany(...).

model/request - Тела запросов: CreateCompanyRequest, CreateUserWithTasksRequest, TaskDto, DoRegisterRequest.

mober2000 commented on src/test/java/com/userbug/CreateCompanyApi.java:
```diff
@@ -0,0 +1,62 @@
+package com.userbug;
```
Api классы должны быть в отдельном классе и не должны быть тестами название класса не соответсвует назначению

mober2000 commented on src/test/java/com/userbug/base/BaseApiTest.java:
```diff
@@ -0,0 +1,13 @@
+package com.userbug.base;
```
Тестовые классы должны быть в отдельной папке не в папке тестов а в папке java/scr

miksenya committed:
Перенес папки с логикой в main/java/pageuserbug
В test/java оставил только тесты и BaseApi с конфигом для тестов.

mober2000 commented on src/test/java/com/userbug/tests/base/BaseApiTest.java:
```diff
@@ -1,4 +1,4 @@
-package com.userbug.base;
```
base api класс так и не перенес из тестов он тут не нужен)))

miksenya replied:
Это конфиг с @Beforeall и сайтом, если его в main переносить то тесты не запустятся, во всех прошлых тестах так же делал.

mober2000 commented:
все конфигурации классы тоже попереноси из тестов во вне

mober2000 commented on src/test/java/com/userbug/tests/CreateCompanyTest.java:
```diff
@@ -0,0 +1,49 @@
+package com.userbug.tests;
```
вот тут все в отдельные классы перенеси очень большая нагромажденная логика нужно покороче сделай методы с переменными в отдельные классы

mober2000 commented on src/test/java/com/userbug/tests/CreateUserWithTasksTest.java:
```diff
@@ -0,0 +1,49 @@
+package com.userbug.tests;
```
тоже самое нагромождено пиздец, смотри как у тебя в юай селение тестах компактно и читабельно сделано и понятно

mober2000 commented on src/test/java/com/userbug/tests/UserRegisterTest.java:
```diff
@@ -0,0 +1,32 @@
+package com.userbug.tests;
```
ну тут тоже самое как и в предыдущих коментах

mober2000 commented on src/main/java/pageuserbug/model/response/DoRegisterResponse.java:
```diff
@@ -0,0 +1,8 @@
+package pageuserbug.model.response;
```
в чем прикол этого класса? где это используется?

miksenya replied:
Это какой-то запасной, если ошибки какие-то будут вроде, но он не обязателен, удалил его

mober2000 commented:
после всех исправлений запусти все тесты что у тебя есть и отправляй обратно на ревью

miksenya committed:
Перенес вообще все что можно из тестов
На все действия создал пошаговые сценарии - пакет Scenarios
В support:
DataGen - генерирует уникальный мэил
CompanySteps - шаги для регистрации компании, что бы, например, тест не падал при регистрации нового пользователя

miksenya committed:
Перенес вообще все что можно из тестов
На все действия создал пошаговые сценарии - пакет Scenarios
В support:
DataGen - генерирует уникальный мэил
CompanySteps - шаги для регистрации компании, что бы, например, тест не падал при регистрации нового пользователя

mober2000 commented:
попереноси все конфиги  из сауса и демблейза из тестов в джава папку тоже

mober2000 commented:
- сделай @description из аллюр для все тестов затем запусти и пришли мне отчет

mober2000 commented on src/test/java/com/userbug/tests/CreateCompanyTest.java:
```diff
@@ -0,0 +1,13 @@
+    @Test
+    void createCompany_minimal() {
+        new CreateCompanyScenario().run(); // все предусловия, вызовы и проверки – внутри
+    }
```
да теперь окей но кода теперь очень мало для одного отдельно класса возьми все тесты юзербагрега и засунь в один чтобы не нагромождать классы
